parsel/selector.py

```diff
@@ -342,6 +354,21 @@ def remove_namespaces(self):
         # remove namespace declarations
         etree.cleanup_namespaces(self.root)
 
+    def remove(self):
+        """
+        Remove matched nodes from the parent element.
+        """
+        try:
+            parent = self.root.getparent()
+        except AttributeError:
+            raise InvalidSelectorError(
```

Comment:
I don't like InvalidSelectorError much, because selector itself is valid, it is just that removing nodes is not supported for this selector. `CantRemoveNodes` or something like this?